
grub2 1.98+20100804-14 (included with squeeze) supports SPA_VERSION up to
(and including) 23.

grub2 1.99~20101210-1 supports SPA_VERSION up to (and including) 28.

zfsutils ATTOW implements SPA_VERSION 28.

--- a/cddl/contrib/opensolaris/cmd/zpool/zpool_main.c
+++ b/cddl/contrib/opensolaris/cmd/zpool/zpool_main.c
@@ -4123,11 +4123,11 @@
 
 	if (cb.cb_poolname[0] != '\0') {
 		(void) printf(
-		    "If you boot from pool '%s', don't forget to update boot code.\n"
-		    "Assuming you use GPT partitioning and da0 is your boot disk\n"
+		    "If you boot from pool '%s', don't forget to update the boot code.\n"
+		    "Assuming you use GNU GRUB and da0 is your boot disk\n"
 		    "the following command will do it:\n"
 		    "\n"
-		    "\tgpart bootcode -b /boot/pmbr -p /boot/gptzfsboot -i 1 da0\n\n",
+		    "\tgrub-install /dev/da0\n\n",
 		    cb.cb_poolname);
 	}
 
--- a/cddl/contrib/opensolaris/lib/libzfs/common/libzfs_pool.c
+++ b/cddl/contrib/opensolaris/lib/libzfs/common/libzfs_pool.c
@@ -2445,11 +2445,11 @@
 			(void) fprintf(stderr, dgettext(TEXT_DOMAIN, "If "
 			    "you boot from pool '%s', you may need to update\n"
 			    "boot code on newly attached disk '%s'.\n\n"
-			    "Assuming you use GPT partitioning and 'da0' is "
+			    "Assuming you use GNU GRUB and 'da0' is "
 			    "your new boot disk\n"
 			    "you may use the following command:\n\n"
-			    "\tgpart bootcode -b /boot/pmbr -p "
-			    "/boot/gptzfsboot -i 1 da0\n\n"),
+			    "\tgrub-install "
+			    "/dev/da0\n\n"),
 			    zhp->zpool_name, new_disk);
 		}
 		return (0);
